src/app/test_executive/zkapps_nonce_test.ml

```diff
@@ -0,0 +1,346 @@
+                     ; account = Nonce (Account.Nonce.of_int 2)
+                     ; valid_while = Ignore
+                     } )
+                []
+            ]
+        in
+        account_updates
+        |> mk_zkapp_command ~memo:"valid zkapp from fish1" ~fee:12_000_000
+             ~fee_payer_pk:fish1_pk ~fee_payer_nonce:(Account.Nonce.of_int 1)
+      in
+      replace_authorizations ~keymap with_dummy_signatures
+    in
+    let%bind.Deferred set_precondition_zkapp_cmd_from_fish1 =
+      let open Zkapp_command_builder in
+      let with_dummy_signatures =
+        let account_updates =
+          mk_forest
+            [ mk_node
+                (mk_account_update_body Signature Call fish1_kp Token_id.default
+                   0
+                   ~update:
+                     { Account_update.Update.dummy with
+                       permissions =
+                         Set { Permissions.user_default with send = Proof }
+                     } )
+                []
+            ]
+        in
+        account_updates
+        |> mk_zkapp_command ~memo:"precondition zkapp from fish1"
+             ~fee:12_000_000 ~fee_payer_pk:fish1_pk
+             ~fee_payer_nonce:(Account.Nonce.of_int 2)
+      in
+      replace_authorizations ~keymap with_dummy_signatures
+    in
+    let%bind.Deferred valid_precondition_zkapp_cmd_from_fish1 =
+      let open Zkapp_command_builder in
+      let with_dummy_signatures =
+        let account_updates =
+          mk_forest
+            [ mk_node
+                (mk_account_update_body Signature Call fish1_kp Token_id.default
+                   0 )
+                []
+            ]
+        in
+        account_updates
+        |> mk_zkapp_command ~memo:"valid zkapp from fish1" ~fee:12_000_000
+             ~fee_payer_pk:fish1_pk ~fee_payer_nonce:(Account.Nonce.of_int 3)
+      in
+      replace_authorizations ~keymap with_dummy_signatures
+    in
+    let snark_work_event_subscription =
+      Event_router.on (event_router t) Snark_work_gossip ~f:(fun _ _ ->
+          [%log info] "Received new snark work" ;
+          Deferred.return `Continue )
+    in
+    let snark_work_failure_subscription =
+      Event_router.on (event_router t) Snark_work_failed ~f:(fun _ _ ->
+          [%log error]
+            "A snark worker encountered an error while creating a proof" ;
+          Deferred.return `Continue )
+    in
+    let%bind () =
+      section
+        "Send a zkapp command with an invalid account update nonce using fish1"
+        (send_zkapp ~logger node invalid_nonce_zkapp_cmd_from_fish1)
+    in
+    let%bind () =
+      section
+        "Send a zkapp command that has its nonce properly incremented after \
+         the fish1 transaction"
+        (send_zkapp ~logger node valid_zkapp_cmd_from_fish1)
+    in
+    let%bind () =
+      section
+        "Wait for fish1 zkapp command with invalid nonce to be rejected by \
```

Comment:
NIT: "rejected by frontier" is odd wording here; the txn is accepted into the frontier, but with a failed status.